### ISSUE TITLE
training issues

### ISSUE THREAD

**Issue by csyufei** [zhaisf/BadT2I#3]
Dear authors of BadT2I

The benign loss and backdoor loss when training are as follows:
![image](https://github.com/zhaisf/BadT2I/assets/103623103/7d7848c9-eb15-400e-8ec9-f94829360f91) 

The x-axis denotes the which epoch. Why is the loss so unstable during training?
Looking forward to your answer in advance.

Best,
Yufei

**Comment by zhaisf** [comment]
Yes, we have also observed similar phenomena. Although I'm not familiar with your specific experimental setup, I'd like to attempt to address your question of unstable loss through the following points:

1. Recall the training loss of Stable Diffusion (Eq. 5): 

    $E_{z,c,\epsilon, t}  \left[ \Vert \epsilon_{\theta} \left(z_t, t, c \right) - \epsilon \Vert_2^2  \right], t \sim \text{Uniform}({1,...,T})$
    
    That is, in each training step,  t  is randomly chosen, leading to a random scale of noise.

2. Our training process builds upon a pretrained stable diffusion model. Fine-tuning for a few steps has little impact on the model's output distribution. Therefore, instability arising from random  noise  leads to more noticeable variations in loss.
